File: hw/villagomez/u2/hw18ZooJS/ec.edu.espe.zoo.model/Mammal.js
const Animal = require('./Animal');

class Mammal extends Animal {
    constructor(id, description, dateOnborn, gender, numberOfBones, cage) {
        if (new.target === Mammal) {
            throw new Error('Cannot instantiate abstract class Mammal');
        }
        super(id, description, dateOnborn, gender, numberOfBones, cage);
    }
}

module.exports = Mammal;

File: ws/ccedeno/u2/WS24Zoo/Zoo/src/main/java/ec/edu/espe/zoo/model/Mammal.java
package ec.edu.espe.zoo.model;

import java.util.Date;

/**
 *
 * @author Andres Cedeno, codecreators, DCCO-ESPE
 */
public abstract class Mammal extends Animal {
    
    public Mammal(int id, String description, Date dateOnborn, char gender, int numberOfBones, Cage cage) {
        super(id, description, dateOnborn, gender, numberOfBones, cage);
    }
    
}
